### PR TITLE
Round setup and battle timers

### PR DESCRIPTION
**Pull request by pygmychameleon**
Set up round by 
1. put monsters at random positions
2. restart the scene

Add timers to start new round when round is over.

### CODE REVIEW

File: client/src/scenes/Battle.ts [hunk 1]
```diff
@@ -7,26 +7,66 @@ export default class Battle extends Phaser.Scene {
...
+        this.time.addEvent({
+            delay: 1000, // 1000 milliseconds = 1 second
+            callback: this.updateTimer,
+            callbackScope: this,
+            loop: true
+        });
     }
 
     async create() {
+        this.cursors = this.input.keyboard.createCursorKeys()
```
Comment (winson8222) [inline client/src/scenes/Battle.ts:68]: Lets standardise to put those that can be loaded, like cursor/keyboard input into the preload?

File: client/src/scenes/Battle.ts [hunk 2]
```diff
@@ -125,10 +156,67 @@ export default class Battle extends Phaser.Scene {
...
+        this.setUpTimer();
     }
 
+    private updateTimer() {
+        this.remainingTime = Math.max(0, this.remainingTime - 1);
+        this.timerText.setText(`Time: ${this.remainingTime}`);
+    }
+
+
+    private setUpMap(xOffset: number, yOffset: number) {
```
Comment (winson8222) [inline client/src/scenes/Battle.ts:180]: There is no collision set with the player, is it omitted on purpose?

File: client/public/enemies/dragon.png
Comment (winson8222) [inline client/public/enemies/dragon.png:1]: beautiful

**Review by winson8222 — CHANGES_REQUESTED** [review]
Thanks for making the random monsters spawn logics. Just wondering if the resolution of the game changed do we need to change all the offsets of the tilemap? I am not sure if after merging mine it will cause the map to explode.

**Comment by yongning0310** [comment]
pending bug fixes

**Pull request closed without merging**